### PR TITLE
port RpcMempool10k from BCHN

### PR DESCRIPTION
**Pull request by Greg-Griffith**
This will come in handy in a short while for benchmarking new UniValue code changes/optimizations.

I added a new benchmark alongside the old "RpcMempool" benchmark (which
creates a tiny 1k mempool with simple tx's in it), called
"RpcMempool10k" which adds a similar test but with a larger mempool of
10k tx's, each tx being more comples (10 ins, 10 outs).

This new benchmark is a better test of UniValue -> JSON, amongst other
things.

The new benchmark takes ~21 seconds on my system to execute.

When the new optimizations that for UniValue are being prepared are in
place, that number will drop down to the expected 5 (all the other tests
take 5 seconds on my system to execute, give or take 1 second).

Note we need this in place in master before my upcoming changes to
UniValue to make testing easier -- you can switch between master and my
patches to characterize the effects of MR's I plan on submitting.

This commit maintains the predicate that we don't modify old benchmarks
for the time being (so as to compare regressions to ABC), and that we
only add new benchmarks alongside existing ones.

### CODE REVIEW

File: src/bench/rpc_mempool.cpp [hunk 1]
```diff
@@ -12,22 +12,23 @@
 #include <list>
 #include <vector>
 
+extern CTxMemPool mempool;
 
-static void AddTx(const CTransactionRef &tx, const CAmount &nFee, CTxMemPool &pool)
```
Comment (sickpig) [inline src/bench/rpc_mempool.cpp:16]: why this change? Is this due to the fact that passing mempool reference to `AddTx` invalidate the validity of this bench?

Comment (Greg-Griffith) [reply src/bench/rpc_mempool.cpp:16]: when passing around a mempool object, due to the way some other functions work, the test always returned an empty json object. using the global mempool object fixes that issue
